docs/web/sdk-service-reference.mdx

```diff
@@ -0,0 +1,239 @@
+---
+title: Service Reference
+description: Geocortex Web - Service Reference
+---
+
+import useBaseUrl from "@docusaurus/useBaseUrl";
+import UseCaseCard from "../../src/components/UseCaseCard";
+import UseCaseContainer from "../../src/components/UseCaseContainer";
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Services act as a repository of accessible behavior and data, and can be used for a variety of purposes, from implementing commands and operations to managing shared data, to interfacing with the REST API of an external service.
+
+## Service Registration
+
+All services need to be registered with Geocortex Web. Services can optionally use the `itemType` property to bind to an app item in config.
+
+```ts
+export default function(registry: LibraryRegistry) {
+    ...
+    registry.registerService({
+        id: "custom-service",
+        getServiceType: () => CustomService,
+        // Use this setting if you want your service to load an
+        // item from the app config
+        itemType: "custom-service-model",
+        // Use this setting if you want your service to load on
+        // application startup. Defaults to `false`.
+        loadOnStartup: true,
+    })
+    ...
+}
+```
+
+## Service Anatomy
+
+All services have access to initialization and cleanup methods, as well as the `messages` property that allows for the registration and execution of commands and operations.
+
+### Initialization and Cleanup
+
+Sometimes, a service is required to perform initialization and cleanup actions, for example, to initialize dynamic data and cleanup dangling references. The `ServiceBase` class has two methods to achieve this:
+
+-   `_onInitialize()`, which can be overridden,
+-   and `destroy()`, which can be implemented to facilitate cleanup.
+
+The following service uses these methods to subscribe to an event and cleanup the event handler afterwards.
+
+:::important
+Always call `super._onInitialize()` before any custom initialization logic.
+:::
+
+```ts
+import { ServiceBase } from "@geocortex/web/services";
+import { MapEvent } from "@geocortex/viewer-framework/messaging/registry/map";
+
+export default class CustomService extends ServiceBase {
+    handles: IHandle[] = [];
+
+    protected async _onInitialize(): Promise<void> {
+        await super._onInitialize();
+        this.handles.push(
+            this.messages.events.map.initialized.subscribe((e: MapEvent) => {
+                console.log(
+                    "Map Initialized, do your map dependent model setup here."
+                );
+            })
+        );
+        return;
+    }
+
+    destroy(): void {
+        this.handles.forEach((h) => h.remove());
+    }
+}
+```
+
+### Message Bus
```

Comment:
I think this heading and content would be better in the next article - `Commands and Operations in Services`. When reading that article i was missing this important info. We may want to briefly message the message bus then link to the other article?

Reply:
Good Idea!! Moved the content to "Commands and Operations in Services", and added a link here